### PR TITLE
Use specific failureDomain while creating new cephblockpool

### PR DESCRIPTION
**Pull request by prsurve**
Fixes: #1067 
Signed-off-by: prsurve <prsurve@redhat.com>

### CODE REVIEW

**Comment by prsurve** [comment]
Verifed Locally
### 1az setup creating cephblockpool with fd as rack
 
```
14:08:10 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding CephBlockPool with name cbp-test-4df1174cf7254a45a899966748513fb8
14:08:11 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: ceph.rook.io/v1
                                                           kind: CephBlockPool
                                                           metadata:
                                                             name: cbp-test-4df1174cf7254a45a899966748513fb8
                                                             namespace: openshift-storage
                                                           spec:
                                                             annotations: null
                                                             failureDomain: rack
                                                             replicated:
                                                               size: 3
14:08:11 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/prsurve/OCS4/1azipi27/auth/kubeconfig create -f /tmp/CephBlockPoolsyyti96m -o yaml

```
### 3az setup creating cephblockpool with fd as host
```
14:10:24 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding CephBlockPool with name cbp-test-8c7da695a54340458ee0bcba16b429e6
14:10:24 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: ceph.rook.io/v1
                                                           kind: CephBlockPool
                                                           metadata:
                                                             name: cbp-test-8c7da695a54340458ee0bcba16b429e6
                                                             namespace: openshift-storage
                                                           spec:
                                                             annotations: null
                                                             failureDomain: zone
                                                             replicated:
                                                               size: 3
14:10:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig create -f /tmp/CephBlockPool5an7a0b0 -o yaml

```

File: tests/helpers.py [hunk 1]
```diff
@@ -1496,3 +1497,20 @@ def create_dummy_osd(deployment):
...
+def get_failure_domin():
+    """
+    Function is used to getting failure domain of pool
+
+    Returns:
+        str: Failure domain from cephblockpool
+
+    """
+    ct_pod = pod.get_ceph_tools_pod()
+    out = ct_pod.exec_ceph_cmd(ceph_cmd="ceph osd crush rule dump", format='json')
```
Comment (ebenahar) [inline tests/helpers.py]: Please examine out and assert in case it's empty

File: tests/helpers.py [hunk 2]
```diff
@@ -1496,3 +1497,21 @@ def create_dummy_osd(deployment):
...
+    assert out, "Failed to get cmd output"
+    for crush_rule in out:
+        if constants.CEPHBLOCKPOOL.lower() in crush_rule.get("rule_name"):
```
Comment (RazTamir) [inline tests/helpers.py]: Do we need some verification for cephfs?

Comment (prsurve) [reply tests/helpers.py]: No, we don't need as for cephfs because failure domain is applicable for cephblockpool only

File: tests/helpers.py [hunk 3]
```diff
@@ -271,6 +271,7 @@ def create_ceph_block_pool(pool_name=None):
         )
     )
     cbp_data['metadata']['namespace'] = defaults.ROOK_CLUSTER_NAMESPACE
+    cbp_data['spec']['failureDomain'] = get_failure_domin()
```
Comment (vasukulkarni) [inline tests/helpers.py]: with my PR for fixing defaults, do we still need this?

Comment (vasukulkarni) [reply tests/helpers.py]: You can ignore until its merged!

**Pull request merged**